akka-bbb-apps/src/main/scala/org/bigbluebutton/core/models/Plugins.scala

```diff
@@ -112,6 +116,26 @@ object PluginModel {
     val baseUrl = plugin.manifest.url.substring(0, plugin.manifest.url.lastIndexOf('/') + 1)
     baseUrl + relativeUrl
   }
+  private def createFinalJavascriptEntrypointUrl(plugin: Plugin, jsEntrypointAbsoluteUrl: String): String = {
+    val maybeVersion = for {
+      manifest <- plugin.manifest.content
+      version  <- manifest.pluginVersion
+      if Version.isValid(version)
+    } yield version
+
+    plugin.manifest.content.foreach { manifest =>
+      manifest.pluginVersion.foreach { version =>
+          if (!Version.isValid(version)) {
+            logger.warn("pluginVersion for [{}] is not valid, ignoring...", manifest.name)
+          }
+        }
+    }
+
+    maybeVersion match {
+      case Some(version) => new URIBuilder(jsEntrypointAbsoluteUrl).setParameter("version", version).toString
+      case None          => jsEntrypointAbsoluteUrl
+    }
+  }
```

Comment:
Let's avoid iterating over the same props twice. Here's a simplified version:

```suggestion
  private def createFinalJavascriptEntrypointUrl(plugin: Plugin, jsEntrypointAbsoluteUrl: String): String = {
    (for {
      manifest <- plugin.manifest.content
      version  <- manifest.pluginVersion
    } yield {
      if(Version.isValid(version, false)) {
        new URIBuilder(jsEntrypointAbsoluteUrl).setParameter("version", version).toString
      } else {
        logger.warn("pluginVersion {} for [{}] is not valid, ignoring...", version, manifest.name)
        jsEntrypointAbsoluteUrl
      }
    }).getOrElse(jsEntrypointAbsoluteUrl)
  }
```